dask/core.py

```diff
@@ -303,8 +304,23 @@ def subs(task, key, val):
             arg = subs(arg, key, val)
         elif type_arg is list:
             arg = [subs(x, key, val) for x in arg]
-        elif type_arg is type(key) and arg == key:
-            arg = val
+        elif type_arg is type(key):
+            # https://github.com/dask/dask/pull/2457
+            # mute a FutureWarning from NumPy that may bubble up here
+            # about "elementwise comparison failed, returning scalar instead"
+            # when doing something like Array(...).compute()
+            # since `arg` will happen to be a NumPy array
+            # In the future this will raise a
+            # ValueError: The truth value of an array with more than one ...
+            # We want to prepare for that, so we catch ValueError, but
+            # we also don't want users seeing the warning
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", FutureWarning)
+                try:
+                    if arg == key:
+                        arg = val
+                except ValueError:
+                    pass
```

Comment:
I'll write and run a benchmark. (FYI, I should have those running regularly later today. Waiting for the pandas backlog to finish first).